core/src/main/scala/org/apache/spark/status/api/v1/OneApplicationResource.scala

```diff
@@ -51,6 +52,46 @@ private[v1] class AbstractApplicationResource extends BaseAppResource {
   @Path("executors")
   def executorList(): Seq[ExecutorSummary] = withUI(_.store.executorList(true))
 
+  @GET
+  @Path("executors/{executorId}/threads")
+  def threadDump(@PathParam("executorId") executorId: String): Response = withUI { ui =>
+    uiRoot.executorThreadDumpsNotAvailableError().getOrElse {
+      val safeExecutorId =
+        Option(UIUtils.stripXSS(executorId)).map { executorId =>
+          UIUtils.decodeURLParameter(executorId)
+        }.getOrElse {
+          throw new IllegalArgumentException(s"Missing executorId parameter")
+        }
+
+      def isAllDigits(x: String) = x.forall(Character.isDigit)
+
+      if (executorId != SparkContext.DRIVER_IDENTIFIER && !isAllDigits(executorId)) {
```

Comment:
Shall we change the "executorId" to low case before doing the comparison?

Reply:
I don't think so. As I see the app-id is also not case insensitive at the URL api/v1/applications/{app-id}.